packages/typespec-client-generator-core/src/types.ts

```diff
@@ -184,80 +186,162 @@ export function addEncodeInfo(
   return diagnostics.wrap(undefined);
 }
 
-/**
- * Mapping of typespec scalar kinds to the built in kinds exposed in the SDK
- * @param scalar the original typespec scalar
- * @returns the corresponding sdk built in kind
- */
-function getScalarKind(scalar: Scalar): SdkBuiltInKinds {
-  if (isSdkBuiltInKind(scalar.name)) {
-    return scalar.name;
+function getScalarTypeKind(context: TCGCContext, type: Scalar): IntrinsicScalarName | "any" {
+  if (context.program.checker.isStdType(type)) {
+    return type.name;
   }
-  throw Error(`Unknown scalar kind ${scalar.name}`);
+
+  // for those scalar defined as `scalar newThing;`,
+  // the best we could do here is return as a `any` type with a name and namespace and let the generator figure what this is
+  if (type.baseScalar === undefined) {
+    return "any";
+  }
+
+  return getScalarTypeKind(context, type.baseScalar);
+}
+
+function getSdkDateTimeType(
+  context: TCGCContext,
+  type: Scalar,
+  kind: "utcDateTime" | "offsetDateTime"
+): [SdkDatetimeType, readonly Diagnostic[]] {
+  const diagnostics = createDiagnosticCollector();
+  const docWrapper = getDocHelper(context, type);
+  const encodeData = getEncode(context.program, type);
+  const baseType = type.baseScalar
+    ? diagnostics.pipe(getSdkDateTimeType(context, type.baseScalar, kind))
+    : undefined;
+  // first we try if this type has an encode on the encode decorator, if not, we get the encode from base type, if still not, we fallback to the default value of datetime rfc3339
+  const encode = encodeData?.encoding ?? baseType?.encode ?? "rfc3339";
```

Comment:
can we keep getting the encode and wire type as a separate function for readability? It's hard to follow with that many question marks

Reply:
Yeah - this might be messy.
I will make a function for this so that it could be more clear

Reply:
This part has been refactored.